solr/core/src/java/org/apache/solr/search/facet/FacetProcessor.java

```diff
@@ -202,6 +202,22 @@ private void handleFilterExclusions() throws IOException {
       return;
     }
 
+    List<Query> qlist = getContextQueriesWithExclusions(excludeSet);
+
+    // recompute the base domain
+    fcontext.base = fcontext.searcher.getDocSet(qlist);
+  }
+
+  protected List<Query> getContextQueries() {
+    Set<Query> excludeSet = QueryUtils.getTaggedQueries(fcontext.req, freq.domain.excludeTags);
+    return getContextQueriesWithExclusions(excludeSet);
+  }
+
+  private List<Query> getContextQueriesWithExclusions(Set<Query> excludeSet) {
+    if (excludeSet == null) {
+      excludeSet = QueryUtils.getTaggedQueries(fcontext.req, freq.domain.excludeTags);
+    }
+
```

Comment:
Nice! but with fb7b94090f91f84eb63b19a064e0f934e98ed69d, we no longer need this check, right?